ShellPkg/Library/UefiShellLib/UefiShellLib.c

```diff
@@ -4232,7 +4239,7 @@ ShellConvertStringToUint64 (
     return (EFI_NOT_FOUND);
   }
 
-  if (Value != NULL) {
+  if ((Value != NULL) && !EFI_ERROR (Status)) {

```

Comment:
I think this conditional should be the other way around. 
```suggestion
  if (!EFI_ERROR (Status) && (Value != NULL)) {

```

Reply:
Is is easier to read this way, because it is the same way as the `if` statement just above it, on line 4238. If you want to change this, you should change both. For maximal execution efficiency of the most "likely" case you may have a subtle point, but it is very minor. If this is run in a busy loop, you have bigger trouble already.

Reply:
Or is it just easier to read that way? Opinions welcome.

Reply:
I think the easiest to read would be to replace this,the preceding if-statement, and the rest of the function with:

```
if (EFI_ERROR (Status)) {
  return EFI_INVALID_PARAMETER;
}

if (Value == NULL) {
  return EFI_NOT_FOUND;
}

*Value = RetVal;

return EFI_SUCCESS;
```

Reply:
Yes, that is much more straight-forward. I was too focused on making a minimal change, and the original authors probably on having a single point of return. With your suggestion it is much clearer how we return one of the documented return values, and no other.

Reply:
I fixed up this commit.